### PR TITLE
fix: throw error and exit when `package.json` not found

### PR DESCRIPTION
**Pull request by harish-sethuraman**
Addresses: 
- https://github.com/eslint/create-config/issues/19

We will early exit when `package.json` is not found

### CODE REVIEW

File: lib/init/config-initializer.js [hunk 1]
```diff
@@ -345,7 +346,12 @@ function askInstallModules(modules, packageJsonExists) {
  * @returns {Promise<void>} The promise with the result of the prompt
  */
 function promptUser() {
+    const packageJsonExists = npmUtils.checkPackageJson();
 
+    if (!packageJsonExists) {
+        error("Cannot initialize ESLint outside a node project");
+        exit();
```
Comment (aladdin-add) [inline lib/init/config-initializer.js:353]: better to give it a non-zero value to state it's not successful.
```suggestion
        exit(1);
```

File: lib/init/config-initializer.js [hunk 2]
```diff
@@ -427,8 +433,7 @@ function promptUser() {
                 { message: "XO: https://github.com/xojs/eslint-config-xo", name: "xo" }
             ],
             skip() {
-                this.state.answers.packageJsonExists = npmUtils.checkPackageJson();
-                return !(this.state.answers.source === "guide" && this.state.answers.packageJsonExists);
+                return !(this.state.answers.source === "guide" && packageJsonExists);
```
Comment (aladdin-add) [inline lib/init/config-initializer.js:436]: if packageJsonExists is alway `true`, so the check seems no longer needed.
```suggestion
                return !(this.state.answers.source === "guide");
```

File: lib/init/config-initializer.js [hunk 3]
```diff
@@ -469,13 +474,13 @@ function promptUser() {
             const config = processAnswers(earlyAnswers);
             const modules = getModulesList(config);
 
-            return askInstallModules(modules, earlyAnswers.packageJsonExists)
+            return askInstallModules(modules, packageJsonExists)
```
Comment (aladdin-add) [inline lib/init/config-initializer.js:477]: same here.

File: lib/init/config-initializer.js [hunk 4]
```diff
@@ -469,13 +474,13 @@ function promptUser() {
...
                 .then(() => writeFile(config, earlyAnswers.format));
         }
 
         // early exit if you are using a style guide
         if (earlyAnswers.source === "guide") {
-            if (!earlyAnswers.packageJsonExists) {
+            if (!packageJsonExists) {
```
Comment (aladdin-add) [inline lib/init/config-initializer.js:483]: same here.

File: lib/init/config-initializer.js [hunk 5]
```diff
@@ -498,7 +503,7 @@ function promptUser() {
 
             const modules = getModulesList(config);
 
-            return askInstallModules(modules, earlyAnswers.packageJsonExists)
+            return askInstallModules(modules, packageJsonExists)
```
Comment (aladdin-add) [inline lib/init/config-initializer.js:506]: same here.

File: lib/init/config-initializer.js [hunk 6]
```diff
@@ -456,7 +461,7 @@ function promptUser() {
             disabled: "No",
             initial: 1,
             skip() {
-                return !(this.state.answers.source === "guide" && this.state.answers.packageJsonExists && hasESLintVersionConflict(this.state.answers));
+                return !(this.state.answers.source === "guide" && packageJsonExists && hasESLintVersionConflict(this.state.answers));
```
Comment (aladdin-add) [inline lib/init/config-initializer.js:464]: same here.

File: lib/init/config-initializer.js (hunk 1 above)
Comment (harish-sethuraman) [reply lib/init/config-initializer.js]: Done

File: lib/init/config-initializer.js (hunk 2 above)
Comment (harish-sethuraman) [reply lib/init/config-initializer.js]: Thanks. I forgot to remove these 😅

**Review by aladdin-add — CHANGES_REQUESTED** [review]
the change LGTM, can you please add a test?

**Review by harish-sethuraman — COMMENTED** [review]
I've removed the process exit and error console and replaced it with `Error` and wrote test for the same

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks! 👍

**Comment by harish-sethuraman** [comment]
Bump

File: lib/init/config-initializer.js [hunk 7]
```diff
@@ -345,7 +344,11 @@ function askInstallModules(modules, packageJsonExists) {
  * @returns {Promise<void>} The promise with the result of the prompt
  */
 function promptUser() {
+    const packageJsonExists = npmUtils.checkPackageJson();
 
+    if (!packageJsonExists) {
+        throw new Error("Cannot initialize ESLint outside a node project");
```
Comment (snitin315) [inline lib/init/config-initializer.js:350]:
```suggestion
        throw new Error("A package.json file is necessary to initialize ESLint. Run `npm init` to create a package.json file and try again.");
```

File: tests/init/config-initializer.js [hunk 8]
```diff
@@ -137,6 +138,21 @@ describe("configInitializer", () => {
...
+                        ...await esmock("../../lib/init/npm-utils.js", { fs: defineInMemoryFs({}) })
+                    }
+                };
+
+
+                init = await esmock("../../lib/init/config-initializer.js", requireStubs, { });
+
+                assert.throws(() => {
+                    init.initializeConfig();
+                }, "Cannot initialize ESLint outside a node project");
```
Comment (snitin315) [inline tests/init/config-initializer.js:154]:
```suggestion
                }, "A package.json file is necessary to initialize ESLint. Run `npm init` to create a package.json file and try again.");
```

Need to update here as well.

**Review by snitin315 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**